### PR TITLE
FEAT: adding cursor.tables 

### PR DESCRIPTION
**Pull request by jahnvi480**
### Work Item / Issue Reference  

> [AB#34926](https://sqlclientdrivers.visualstudio.com/c6d89619-62de-46a0-8b46-70b92a84d85e/_workitems/edit/34926)

-------------------------------------------------------------------
### Summary   
This pull request adds a new `tables()` method to the `Cursor` class in `mssql_python/cursor.py`, providing a way to query metadata about tables in the database, including support for filtering by name, schema, catalog, and table type. It also introduces comprehensive test coverage for this new method in `tests/test_004_cursor.py`. Additionally, the `skip()` method in the cursor is simplified by delegating to the existing `scroll()` method.

**New feature: Table metadata querying**
- Added a `tables()` method to the `Cursor` class, enabling users to retrieve information about tables with support for filtering by table name (including temporary tables), schema, catalog, and table type (supports both string and list input). The method returns the cursor itself for easy chaining and iteration.

**Testing improvements**
- Introduced a suite of tests for the new `tables()` method, covering basic usage, filtering by name, schema, and type, wildcard support, combined filters, empty results, iteration, method chaining, and existence checks. These tests ensure the method works as intended and handles edge cases.

**Code simplification**
- Refactored the `skip()` method in the cursor to delegate to the `scroll()` method in 'relative' mode, removing redundant validation and manual row skipping logic.

### CODE REVIEW

File: mssql_python/pybind/ddbc_bindings.cpp [hunk 1]
```diff
@@ -982,6 +984,93 @@ SQLRETURN SQLExecDirect_wrap(SqlHandlePtr StatementHandle, const std::wstring& Q
     return ret;
 }
 
+// Wrapper for SQLTables
+SQLRETURN SQLTables_wrap(SqlHandlePtr StatementHandle, 
+                         const std::wstring& catalog,
+                         const std::wstring& schema, 
+                         const std::wstring& table,
+                         const std::wstring& tableType) {
+    LOG("Calling SQLTables API with parameters: catalog={}, schema={}, table={}, tableType={}",
```
Comment (sumitmsft) [inline mssql_python/pybind/ddbc_bindings.cpp]: The LOG macros in `SQLTables_wrap` log parameters such as catalog, schema, table, and tableType. While useful for debugging, these may contain sensitive information. Please ensure that in production or default builds, LOG statements do not expose sensitive data, or add a mechanism to redact/log at an appropriate level.

File: mssql_python/cursor.py [hunk 2]
```diff
@@ -1258,19 +1261,134 @@ def skip(self, count: int) -> None:
...
+        
+        # Initialize description from column metadata
+        column_metadata = []
+        try:
+            ddbc_bindings.DDBCSQLDescribeCol(self.hstmt, column_metadata)
+            self._initialize_description(column_metadata)
+        except Exception:
+            # If describe fails, create a manual description for the standard columns
+            column_types = [str, str, str, str, str]
+            self.description = [
```
Comment (sumitmsft) [inline mssql_python/cursor.py:1373]: In the tables() method, only the standard five columns (table_cat, table_schem, table_name, table_type, remarks) are processed, and any extra columns returned by the driver are ignored. This is a safe and compatible approach, but it would be helpful to `document this behavior` in the method docstring or code comments to avoid confusion for users or future maintainers.

Comment (jahnvi480) [reply mssql_python/cursor.py:1373]: Adding docstring in this method as well as will add this behavior in our Wiki page

**Review by sumitmsft — APPROVED** [review]

**Review by gargsaumya — APPROVED** [review]
